src/Modal/Modal.component.tsx

```diff
@@ -18,73 +26,90 @@ import { ModalClose } from './Close';
 
 export type ModalSize = 'lg' | 'sm';
 
-const Sizes = {
-    lg: { width: 50, contentPadding: 2.5, footerHeight: 6.5 },
-    sm: { width: 25, contentPadding: 2, footerHeight: 5.5 },
+export const MODAL_THEME_KEY = 'modal';
+export const MODAL_THEME = {
+    size: {
+        lg: {
+            width: 50,
+            contentPadding: 2.5,
+            footerHeight: 6.5,
+            height: 42.375,
+        },
+        sm: {
+            width: 25,
+            contentPadding: 2,
+            footerHeight: 5.5,
+            height: 42.375,
+        },
+    },
 };
 
-const defaultSize: ModalSize = 'lg';
+const size = variant({
+    themeKey: MODAL_THEME_KEY,
+    prop: 'size',
+    variants: MODAL_THEME.size,
+    defaultValue: 'lg',
+});
 
 const StyledModal = StyledReactModal.default.styled`
     position: relative;
-    width: ${({ width, size = defaultSize }: ModalProps) =>
-        width || `${Sizes[size].width}rem`};
-    height: ${({ height = '42.375rem' }: ModalProps) => height};
-    margin: ${({ margin = 'auto' }: ModalProps) => margin};
+
+    ${compose(
```

Comment:
This could be written alternatively like
```tsx
${marginStyles}
${backgroundStyles}
${colorStyles}
${borderRadiusStyles}
```

I wrote it with `compose` because Styled System recommends it for better performance of multiple styles https://styled-system.com/api#compose

Reply:
Nah, I saw the docs on this and like the approach you took.